hamcrest/Hamcrest/File/IsExistingFile.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @author Vasek Brychta <vaclav@brychtovi.cz>
+ */
+
+namespace Hamcrest\File;
+
+use Hamcrest\BaseMatcher;
+use Hamcrest\Description;
+
+class IsExistingFile extends FileMatcher
+{
+	public function __construct()
+	{
+		parent::__construct('an existing file', 'is not a file');
+	}
+
+	protected function matchesFile(\SplFileInfo $file): bool
+	{
+		return $file->isFile();
+	}
+
+	/**
+	 * Evaluates to true if the file exists and is a regular file.
+	 * Accepts only <code>\SplFileInfo</code> objects or <code>string</code> paths.
+	 *
+	 * @return \Hamcrest\File\IsExistingFile
+	 * @factory
```

Comment:
This is likely to fail when used by a generator mentioned above.

Reply:
Why do you think it might fail? I used the generator and it worked correctly.

Reply:
Because I saw `@factory` annotation elsewhere with a parameter (e.g. `@factory la-la`) and assumed (haven't checked with generator code), that generator might fail.